pulpcore/tasking/worker.py

```diff
@@ -392,6 +403,55 @@ def handle_available_tasks(self):
                 keep_looping = True
                 self.supervise_task(task)
 
+    def record_unblocked_waiting_tasks_metric(self):
+        if self.last_metric_heartbeat:
+            worker_ttl_delta = timedelta(seconds=(settings.WORKER_TTL / 3))
+            if timezone.now() > self.last_metric_heartbeat + worker_ttl_delta:
+                try:
+                    with PGAdvisoryLock(TASK_METRICS_HEARTBEAT_LOCK):
+                        meter = get_meter(__name__)
+                        unblocked_waiting_tasks_meter = meter.create_gauge(
+                            name="tasks_unblocked_waiting_queue",
+                            description="Number of unblocked tasks waiting in the queue.",
+                            unit="tasks",
+                        )
+
+                        longest_waiting_tasks_meter = meter.create_gauge(
+                            name="tasks_longest_waiting_task",
+                            description="The age of the longest waiting task.",
+                            unit="seconds",
+                        )
+
+                        unblocked_waiting_tasks = Task.objects.filter(
+                            unblocked_at__isnull=False, started_at__isnull=True
+                        )
+                        unblocked_waiting_tasks_annotated = unblocked_waiting_tasks.annotate(
+                            unblocked_for=Value(timezone.now()) - F("unblocked_at")
+                        )
+                        unblocked_waiting_tasks_aggregated = (
+                            unblocked_waiting_tasks_annotated.aggregate(
+                                longest_age=Coalesce(Max("unblocked_for"), timezone.timedelta(0)),
```

Comment:
```suggestion
                                longest_age=Max("unblocked_for", default=timezone.timedelta(0),
```
Is this one possible?

Reply:
I think it is. Gonna test it before committing your suggestion.